### PR TITLE
bsp: stm32: disable stm32-rng in favor of optee-rng

### PR DESCRIPTION
**Pull request by igoropaniuk**
Disable HW_RANDOM_STM32 in favor of HW_RANDOM_OPTEE.

In the documentation for this SoC [1] there is a statement that RNG1 instance should be assigned to OP-TEE "OR" Linux, but not to both. In our setup OP-TEE is handling STM32 RNG peripheral already.

[1] https://wiki.st.com/stm32mpu/wiki/RNG_internal_peripheral#On_STM32MP15x_lines_2

### CODE REVIEW

**Review by ricardosalveti — APPROVED** [review]
LGTM, please also propose to main after this gets merged.

**Comment by igoropaniuk** [comment]
@ricardosalveti https://github.com/foundriesio/lmp-kernel-cache/pull/145

**Pull request merged**
